### PR TITLE
refactor(TextInput): support right inner content

### PR DESCRIPTION
**Pull request by korvin89**

### CODE REVIEW

File: src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -150,45 +148,44 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
...
-                    {`${label}`}
-                </label>
-            )}
-            {multiline ? (
-                <TextAreaControl {...props} {...commonProps} controlRef={handleRef} />
-            ) : (
-                <InputControl {...props} {...commonProps} controlRef={handleRef} />
-            )}
-
+            <span className={b('content')}>
```
Comment (amje) [inline src/components/TextInput/TextInput.tsx:169]: Do we actually need this extra wrapper?

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -150,45 +148,44 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
...
+                    </label>
+                )}
+                {multiline ? (
+                    <TextAreaControl {...props} {...commonProps} controlRef={handleRef} />
+                ) : (
+                    <InputControl {...props} {...commonProps} controlRef={handleRef} />
+                )}
+                {isClearControlVisible && (
+                    <ClearAction
+                        className={modsClassName(b('clear', {textarea: true}))}
```
Comment (amje) [inline src/components/TextInput/TextInput.tsx:182]: always true?

File: src/components/TextInput/TextInput.scss [hunk 3]
```diff
@@ -91,22 +104,23 @@ $block: '.#{variables.$ns}text-input';
...
 
-    /* increase css specificity to rewrite Button styles */
-    &__clear[class] {
-        --yc-button-background-color: transparent;
-        --yc-button-background-color-hover: transparent;
+    &__clear {
+        /* increase css specificity to rewrite Button styles */
+        &[class] {
+            --yc-button-background-color: transparent;
+            --yc-button-background-color-hover: transparent;
```
Comment (amje) [inline src/components/TextInput/TextInput.scss:111]: How about not using Button component here? Let it be just `<button>` with styling. These vars are private for button

File: src/components/TextInput/__tests__/TextInput.input.test.tsx [hunk 4]
```diff
@@ -29,12 +29,6 @@ describe('TextInput input', () => {
                 expect(container.querySelector('.yc-text-input__error')).not.toBeInTheDocument();
             });
 
-            test('render clear button with hasClear prop', () => {
```
Comment (amje) [inline src/components/TextInput/__tests__/TextInput.input.test.tsx:32]: Why deleted test?

File: src/components/TextInput/__tests__/TextInput.input.test.tsx [hunk 5]
```diff
@@ -74,19 +70,6 @@ describe('TextInput input', () => {
 
                 expect(onChangeFn).toBeCalled();
             });
-
```
Comment (amje) [inline src/components/TextInput/__tests__/TextInput.input.test.tsx:77]: Why deleted test?

File: src/components/TextInput/TextInput.tsx [hunk 6]
```diff
@@ -33,7 +30,8 @@ const prepareAutoComplete = (autoComplete: TextInputProps['autoComplete']): stri
     }
 };
 
-export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(function TextInput(
+// eslint-disable-next-line complexity
+export const TextInput = React.forwardRef<HTMLElement, TextInputProps>(function TextInput(
```
Comment (amje) [inline src/components/TextInput/TextInput.tsx:34]: It can break typings. Let's keep it span for now.

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/TextInput/TextInput.tsx (hunk 1 above)
Comment (korvin89) [reply src/components/TextInput/TextInput.tsx:169]: Unfortunately, yes. This wrap separates input content from error message

File: src/components/TextInput/TextInput.tsx (hunk 2 above)
Comment (korvin89) [reply src/components/TextInput/TextInput.tsx]: Nope =/

File: src/components/TextInput/TextInput.scss (hunk 3 above)
Comment (korvin89) [reply src/components/TextInput/TextInput.scss:111]: I`ve created issue https://github.com/gravity-ui/uikit/issues/645

File: src/components/TextInput/TextInput.tsx [hunk 7]
```diff
@@ -150,45 +148,44 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
     };
 
     return (
-        <span
-            ref={ref}
+        <div
+            ref={ref as React.ForwardedRef<HTMLDivElement>}
```
Comment (amje) [inline src/components/TextInput/TextInput.tsx:152]: Let’s not change to span, it might break someone types

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**